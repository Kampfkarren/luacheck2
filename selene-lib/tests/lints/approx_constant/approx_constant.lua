local good = 3
local good = 3.1
local good = 3.13
local good = 3.15
local good = -3.15
local good = 3.1417
local good = 3.14159266

local good = 0x314
local good = 3_14

local bad = 3.14
local bad = 3.141
local bad = -3.141
local bad = 3.142
local bad = 3.1415
local bad = 3.14159265

local bad = 3.14 + 1
local bad = f(3.14 + 1)
local bad = f(-3.14 + 1)
