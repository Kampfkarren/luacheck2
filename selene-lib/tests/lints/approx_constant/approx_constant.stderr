error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:12:13
   │
12 │ local bad = 3.14
   │             ^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:13:13
   │
13 │ local bad = 3.141
   │             ^^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:14:14
   │
14 │ local bad = -3.141
   │              ^^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:15:13
   │
15 │ local bad = 3.142
   │             ^^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:16:13
   │
16 │ local bad = 3.1415
   │             ^^^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:17:13
   │
17 │ local bad = 3.14159265
   │             ^^^^^^^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:19:13
   │
19 │ local bad = 3.14 + 1
   │             ^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:20:15
   │
20 │ local bad = f(3.14 + 1)
   │               ^^^^

error[approx_constant]: `math.pi` is more precise
   ┌─ approx_constant.lua:21:16
   │
21 │ local bad = f(-3.14 + 1)
   │                ^^^^

